### PR TITLE
Add test for deleting rhrepo

### PR DESCRIPTION
**Pull request by sambible**
Test for BZ 1152672

### CODE REVIEW

File: tests/foreman/ui/test_repository.py [hunk 1]
```diff
@@ -839,6 +839,49 @@ def test_positive_delete_random_docker_repo(session, module_org):
             assert session.repository.search(product_name, repo_name)[0]['Name'] == repo_name
 
 
+@pytest.mark.tier2
+def test_positive_delete_rhrepo(session, module_org):
```
Comment (latran) [inline tests/foreman/ui/test_repository.py:843]:
```suggestion
def test_positive_delete_rhel_repo(session, module_org):
```

File: tests/foreman/ui/test_repository.py [hunk 2]
```diff
@@ -839,6 +839,49 @@ def test_positive_delete_random_docker_repo(session, module_org):
...
+    """Enable and sync a Red Hat Repository, and then delete it
+
+    :id: e96f369d-3e58-4824-802e-0b7e99d6d207
+
+    :expectedresults: Repository can be successfully deleted
+
+    :CaseLevel: Integration
+
```
Comment (latran) [inline tests/foreman/ui/test_repository.py:851]: missing customer case docstring:

`:customerscenario: true`

**Review by latran — CHANGES_REQUESTED** [review]

**Comment by latran** [comment]
trigger: robottelo
pytest: tests/foreman/ui/test_repository.py -k test_positive_delete_rhrepo

**Comment by sambible** [comment]
trigger: robottelo
pytest: tests/foreman/ui/test_repository.py -k test_positive_delete_rhel_repo

File: tests/foreman/ui/test_repository.py [hunk 3]
```diff
@@ -839,6 +839,51 @@ def test_positive_delete_random_docker_repo(session, module_org):
             assert session.repository.search(product_name, repo_name)[0]['Name'] == repo_name
 
 
+@pytest.mark.tier2
+def test_positive_delete_rhel_repo(session, module_org):
```
Comment (latran) [inline tests/foreman/ui/test_repository.py:843]: You passed in `module_org`, but I don't see it being used.  Instead, I see you also called `entities.Organization().create()`.

Use one or the other.  In this instance, use `module_org`

File: tests/foreman/ui/test_repository.py [hunk 4]
```diff
@@ -839,6 +839,51 @@ def test_positive_delete_random_docker_repo(session, module_org):
...
+
+    :customerscenario: true
+
+    :expectedresults: Repository can be successfully deleted
+
+    :CaseLevel: Integration
+
+    :BZ: 1152672
+    """
+
```
Comment (latran) [inline tests/foreman/ui/test_repository.py:856]: Delete whitespace

Comment (latran) [inline tests/foreman/ui/test_repository.py:856]: Delete whitespace

**Review by latran — CHANGES_REQUESTED** [review]

File: tests/foreman/ui/test_repository.py [hunk 5]
```diff
@@ -839,6 +839,51 @@ def test_positive_delete_random_docker_repo(session, module_org):
...
+                    sat_tools_repo.data['arch'],
+                    repository_name,
+                )
+            ]
+        )
+        assert results and all([result == 'Syncing Complete.' for result in results])
+        session.repository.delete(sat_tools_repo.data['product'], repository_name)
+        assert not session.redhatrepository.search(
+            f'name = "{repository_name}"', category='Enabled'
+        )
```
Comment (pondrejk) [inline tests/foreman/ui/test_repository.py:884]: reading the bz, if the product contained just a single repo, it should be cleaned up after repo removal, maybe worth asserting for that too

**Review by pondrejk — APPROVED** [review]
ack, minor comment

**Review by ColeHiggins2 — APPROVED** [review]
ACK, Pending @latran comment

**Review by latran — APPROVED** [review]
ACK

**Pull request merged**